### PR TITLE
Move the ajax form bits of 2 step signin into .ajax-step-flow

### PR DESCRIPTION
**Pull request by walaura**
`ajax-step-flow` will also be used in the request token page so this decouples that from two step signin into its own component. This is a major win for code readability too as everything is now more neatly separated.

The changes are mostly renaming but there's some areas of note:

- Removed `two-step-signin-container` and `two-step-signin-form_section` as they were unused
- Removed `components/two-step-signin/scriptFile.scala` which was an unused editor mistake
- Refactored the back link behaviour in `ajax-step-flow__slide` so it's more reusable instead of a direct mention to the reset link

### CODE REVIEW

File: public/components/ajax-form/ajax-form__slide.js [hunk 1]
```diff
@@ -96,7 +104,7 @@ const fetchSlide = (action, $stateable, fetchProps) =>
       });
     });
 
-const catchSlide = ($stateable, err) => {
```
Comment (jfsoul) [inline public/components/ajax-form/ajax-form__slide.js]: `stateable` doesn't mean much to me.  What is it supposed to be?

File: public/components/ajax-form/ajax-form__slide.js [hunk 2]
```diff
@@ -109,52 +117,62 @@ const catchSlide = ($stateable, err) => {
...
+    .filter(_ =>
+      validAjaxLinkRoutes.map(r => _.href.contains(r)).some(c => c === true)
+    );
+
+  const $forms: HTMLFormElement[] = [
+    ...($component.querySelectorAll(`form`): any)
+  ]
+    .filter(_ => _ instanceof HTMLFormElement)
+    .filter(_ =>
+      validAjaxFormRoutes.map(r => _.action.contains(r)).some(c => c === true)
```
Comment (jfsoul) [inline public/components/ajax-form/ajax-form__slide.js]: is the same as `validAjaxFormRoutes.some(r => _.action.contains(r))`?

**Comment by walaura** [comment]
@jfsoul it really aims to be a pluggable thing you can do to any form. In reality this will be used for signin, potentially signup, and the magic link sender tool (which will become part of the signin flow) so i'm happy to rename it but i'm not sure what to

File: public/components/ajax-form/ajax-form__slide.js (hunk 2 above)
Comment (walaura) [reply public/components/ajax-form/ajax-form__slide.js]: nice catch!

File: public/components/ajax-form/ajax-form__slide.js (hunk 1 above)
Comment (walaura) [reply public/components/ajax-form/ajax-form__slide.js]: i don't know either which is a very telling sign of a bad name, i'll check on it

**Comment by jfsoul** [comment]
Re: `ajax-form` - let's leave it as is for now, but maybe think about putting in a short comment at the top of the file explaining what it's for

**Pull request merged**